src/main/java/fi/aalto/cs/apluscourses/presentation/CourseSelectionViewModel.java

```diff
@@ -0,0 +1,37 @@
+package fi.aalto.cs.apluscourses.presentation;
+
+import fi.aalto.cs.apluscourses.utils.observable.ObservableProperty;
+import fi.aalto.cs.apluscourses.utils.observable.ObservableReadWriteProperty;
+import fi.aalto.cs.apluscourses.utils.observable.ValidationError;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+
+public class CourseSelectionViewModel {
+
+  private List<CourseItemViewModel> courses;
+
+  public final ObservableProperty<String> selectedCourseUrl
+      = new ObservableReadWriteProperty<>("", CourseSelectionViewModel::validateCourseUrl);
+
+  public CourseSelectionViewModel(List<CourseItemViewModel> courses) {
+    this.courses = courses;
+  }
+
+  public CourseItemViewModel[] getCourses() {
+    return courses.toArray(new CourseItemViewModel[0]);
```

Comment:
Should we store `courses` as an array?

Reply:
Yeah good idea, not used as a list anywhere anyways